aea/protocols/generator.py

```diff
@@ -76,6 +82,79 @@ def _extract_all_contents(self) -> Dict[str, Dict[str, str]]:
                 all_contents[performative][content_name] = content_type
         return all_contents
 
+    def _specification_type_to_python_type(self, specification_type: str) -> str:
+        """
+        Convert a data type in protocol specification into its Python equivalent.
+
+        :param specification_type: the type described in the specification
+        :return: The Python equivalent
+        """
+        python_type = ""
+        if specification_type.startswith("pt:optional"):
+            element_type = specification_type[
+                specification_type.index("[") + 1 : specification_type.rindex("]")
+            ]
+            element_type_in_python = self._specification_type_to_python_type(
+                element_type
+            )
+            python_type = "Optional[{}]".format(element_type_in_python)
+        else:
+            specification_types = set(specification_type.split(" or "))
+            if len(specification_types) == 1:  # just one type (not a Union[])
+                if specification_type == "pt:bytes":
```

Comment:
seems like the next few lines could be significantly shortened by:
if specification_type in ["pt:bytes", ...., "pt:str"]
    python_type = specification_type[3:]

Reply:
Done